src/javascript/Edit/Edit.jsx

```diff
@@ -39,13 +46,31 @@ export const EditCmp = ({
                 sections,
                 values
             },
-            editCallback: nodeUuid => {
+            editCallback: (node, mutateNode) => {
                 if (values[Constants.systemFields.OVERRIDE_SUBMIT_CALLBACK]) {
                     values[Constants.systemFields.OVERRIDE_SUBMIT_CALLBACK]();
                 } else {
                     const envEditCallback = envEditCallbacks[contentEditorConfigContext.env];
                     if (envEditCallback) {
-                        envEditCallback(nodeUuid, contentEditorConfigContext);
+                        envEditCallback(node.uuid, contentEditorConfigContext);
+                    }
+                }
+
+                if (mutateNode.rename) {
+                    if (storedLocation && storedLocation.location) {
+                        const newNameStartPosition = mutateNode.rename.lastIndexOf('/') + 1;
+                        const newName = mutateNode.rename.substring(newNameStartPosition, mutateNode.rename.length);
+                        if (openPaths) {
+                            dispatch(replaceOpenedPath(openPaths.map(openPath => openPath.replace(node.name, newName))));
+                        }
+
+                        setStoredLocation({
+                            ...storedLocation,
+                            location: {
+                                ...storedLocation.location,
+                                pathname: storedLocation.location.pathname.replace(node.name, newName)
```

Comment:
Dangerous, you may rename an element of the jcontent that should not be renamed, see if you can export jcontent fct in the registry to be able to extract jcontent path info from the current location and replace the path to rebuild the jcontent url.

Reply:
Take a closer look this two fcts:
**extractParamsFromUrl** and **buildUrl** from **JContent.redux.js**